### PR TITLE
Change and enable strand dump tests to wait on logs

### LINKED ISSUE #38737: [Improvement]: Change strand dump tests to wait on logs instead of timed wait
### Description

$title

Then it is will adapt to delays in starting processes if needed, and will complete the test as soon as possible.

### Describe your problem(s)

_No response_

### Describe your solution(s)

_No response_

### Related area

-> Runtime

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by Nadeeshan96**
$title
Fixes #38737

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/context/LogLeecher.java [hunk 1]
```diff
@@ -42,27 +40,41 @@ public LogLeecher(String text) {
...
     }
+
     /**
      * Feed a log line to check if it matches the expected text.
      *
      * @param logLine The log line which was read
      */
     void feedLine(String logLine) {
-        if (logLine.contains(text)) {
+        if ((isRegExp && logLine.matches(text)) || (!isRegExp && logLine.contains(text))) {
```
Comment (HindujaB) [inline tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/context/LogLeecher.java:77]: We can refactor this to a new method for readability. 
```java
private boolean isTextFound(String logLine) {
        return isRegExp ? logLine.matches(text) : logLine.contains(text);
}
```

File: tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/troubleshoot/StrandDumpTest.java [hunk 2]
```diff
@@ -145,50 +147,52 @@ private void startProcessAndVerifyStrandDump(ProcessBuilder processBuilder, Map<
...
+        try (Stream<String> fileLines = Files.lines(filePath)) {
+            return fileLines.filter(s -> !s.isBlank()).collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new BallerinaTestException("Failure to read from the file: " + filePath);
         }
-        return output;
+    }
+
+    private static boolean isWindowsOS() {
+        return Utils.getOSName().toLowerCase(Locale.ENGLISH).contains("windows");
```
Comment (HindujaB) [inline tests/jballerina-integration-test/src/test/java/org/ballerinalang/test/troubleshoot/StrandDumpTest.java:196]: We have similar methods used in other classes as well. (Eg. `BMainInstance`, `BServerInstance` etc.) Shall we move this to the `Utils` class and reuse it?

**Review by HindujaB — APPROVED** [review]
LGTM

**Review by warunalakshitha — APPROVED** [review]
LGTM

**Pull request merged**